elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -71,14 +81,539 @@ describe('peerPool', () => {
+		let _peerSelectForSendStub: any;
+		let sendToPeer: any;
+
+		beforeEach(async () => {
+			_peerSelectForSendStub = sandbox
+				.stub(peerPool as any, '_peerSelectForSend')
+				.returns([peerInfo]);
+			sendToPeer = sandbox.stub(peerPool, 'sendToPeer').resolves();
+		});
+
+		it('should call _peerSelectForSend', async () => {
+			await peerPool.send(messagePacket);
+
+			expect(_peerSelectForSendStub).to.be.calledOnce;
+		});
+
+		it('should call sendToPeer for each selected peer', async () => {
+			await peerPool.send(messagePacket);
+
+			expect(sendToPeer).to.be.calledOnceWithExactly(
+				messagePacket,
+				constructPeerIdFromPeerInfo(peerInfo),
+			);
+		});
+
+		it(`should emit event if sendToPeer fails`, async () => {
+			sendToPeer.throws();
+			try {
+				await peerPool.send(1 as any);
+			} catch (error) {}
+
+			expect(peerPool.emit).to.be.calledOnce;
+		});
+	});
+
+	describe('#requestFromPeer', () => {
+		it('should throw error if no peers in peerPool', async () => {
+			(peerPool as any)._peerMap = new Map();
+			let err;
+			try {
+				await peerPool.requestFromPeer(requestPacket, peerId);
+			} catch (error) {
+				err = error;
+			}
+
+			expect(err).to.exist;
+		});
+
+		it('should call peer request with packet', async () => {
+			(peerPool as any)._peerMap = new Map([['127.0.0.1:5000', peerObject]]);
+			await peerPool.requestFromPeer(requestPacket, peerId);
+
+			expect(peerObject.request).to.be.calledWithExactly(requestPacket);
+		});
+	});
+
+	describe('#sendToPeer', () => {
+		it('should throw error if no peers in peerPool', async () => {
+			(peerPool as any)._peerMap = new Map();
+			let err;
+			try {
+				await peerPool.sendToPeer(messagePacket, peerId);
+			} catch (error) {
+				err = error;
+			}
+
+			expect(err).to.exist;
+		});
+
+		it('should call peer send with message packet', async () => {
+			const peerStub = {
+				send: sandbox.stub(),
+			};
+			(peerPool as any)._peerMap = new Map([[peerId, peerStub]]);
+			await peerPool.sendToPeer(messagePacket, peerId);
+
+			expect(peerStub.send).to.be.calledWithExactly(messagePacket);
+		});
+	});
+
+	describe('#addInboundPeer', () => {
+		let getPeersStub: any;
+
+		beforeEach(async () => {
+			getPeersStub = sandbox.stub(peerPool, 'getPeers').returns([peerObject]);
+		});
+
+		it('should call getPeers with InboundPeer class', async () => {
+			peerPool.addInboundPeer(peerInfo, peerObject as any);
+
+			expect(getPeersStub).to.be.calledWithExactly(InboundPeer);
+		});
+
+		it('should call _evictPeer if max inbound connections reached', async () => {
+			(peerPool as any)._maxInboundConnections = 0;
+			sandbox.stub(peerPool as any, '_evictPeer');
+			peerPool.addInboundPeer(peerInfo, peerObject as any);
+
+			expect((peerPool as any)._evictPeer).to.be.calledWithExactly(InboundPeer);
+		});
+
+		it('should throw error if peer already exists in peerpool', async () => {
+			(peerPool as any)._peerMap = new Map([[peerId, peerObject]]);
+			let err;
+			try {
```

Comment:
Please, try to remove every `try/catch` as I've proposed above.

Reply:
In this case it is required as `toBeRejectedWith` and `throws` does not work.